### PR TITLE
Re-register UTXO-set-scans after UpdatePool command

### LINKED ISSUE #110: [200 SigmaUSD] Update registered scans and re-scan the wallet after UpdatePool command
Discovered in https://github.com/ergoplatform/oracle-core/pull/109#discussion_r968025042

Since we need to do this on every oracle after they receive a new config, I think we need to check if existing scans reference a changed token id/address/etc. and only then update the scanids.json and trigger a wallet re-scan.

### PR DESCRIPTION
**Pull request by kettlebell**
Close #110

### CODE REVIEW

File: core/src/oracle_state.rs [hunk 1]
```diff
@@ -529,8 +531,8 @@ pub fn register_and_save_scans() -> std::result::Result<(), Error> {
     let datapoint_contract_address =
         OracleContract::checked_load(&config.oracle_box_wrapper_inputs.contract_inputs)?
             .ergo_tree();
-    // If scanIDs.json exists, skip registering scans & saving generated ids
-    if !Path::new("scanIDs.json").exists() {
+
+    let register_scans = || -> std::result::Result<(), Error> {
```
Comment (greenhat) [inline core/src/oracle_state.rs:535]: Extracting this closure into a function would improve `register_and_save_scans()` overall readability.

**Review by greenhat — CHANGES_REQUESTED** [review]
Looking great! Just a minor note.

File: core/src/oracle_state.rs (hunk 1 above)
Comment (kettlebell) [reply core/src/oracle_state.rs]: Sure, it's done.

File: core/src/oracle_state.rs [hunk 2]
```diff
@@ -635,3 +588,68 @@ pub fn register_and_save_scans() -> std::result::Result<(), Error> {
     }
     Ok(())
 }
+
+/// Registers and saves scans to `scanIDs.json` as well as performing wallet rescanning.
+///
+/// WARNING: will overwrite existing `scanIDs.json`!
+fn register_and_save_scans_inner(
+    oracle_pool_participant_token_id: &TokenId,
```
Comment (greenhat) [inline core/src/oracle_state.rs:596]: Do we need these arguments at all? I mean, they are all loaded from oracle config at the call site and are not used anywhere else. I think they could all be defined inside this function.

**Review by greenhat — CHANGES_REQUESTED** [review]
Thank you! Just one tiny comment. :)

File: core/src/oracle_state.rs (hunk 2 above)
Comment (kettlebell) [reply core/src/oracle_state.rs]: Oh yes of course, sorry about that. It's done.

**Pull request merged**